server/repo-op.c

```diff
@@ -1706,30 +1709,58 @@ put_dirent_and_commit (SeafRepo *repo,
                        const char *path,
                        SeafDirent *dent,
                        int replace,
+                       int file_num,
                        const char *user,
                        GError **error)
```

Comment:
You should change the function declaration to: 
 static int
 put_dirent_and_commit (SeafRepo *repo,
                         const char *path,
                         SeafDirent *dents,
                         int n_dents,
                         int replace,
                         const char *user,
                         GError **error)

To make it more suitable to be called by two cases.